### PR TITLE
object-manipulation-2 | Samuel Cho

### PR DESCRIPTION
**Pull request by SamuelCho1**
JavaScript object manipulation practice with unit tests.

### CODE REVIEW

**Comment by SamuelCho1** [comment]
Screen Shot 2021-02-07 at 2 16 36 PM

File: object-manipulation-2/src/omit.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      omitObject[i] = source[i];
+    }
+
+    for (var keysIndex = 0; keysIndex < keys.length; keysIndex++) {
+      for (var j in omitObject) {
+        if (keys[keysIndex] === j) {
+          delete omitObject[j];
+        }
+      }
+    }
```
Comment (thebearingedge) [inline object-manipulation-2/src/omit.js:16]: Try this approach to simplify things a bit:

- for each `sourceKey` in `source`
  - for each `keys` at `i`
    - declare a boolean for whether this `sourceKey` is `toBeOmitted`
    - if `sourceKey` is equal to `keys` at `i`, then
      - this `sourceKey` is `toBeOmitted`
      - `break` out of this inner loop
  - if `sourceKey` is not `toBeOmitted`, then
    - assign `source` at `sourceKey` to your new object at `sourceKey`.

File: object-manipulation-2/src/pick.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* exported pick */
+function pick(source, keys) {
+  var pickObject = {};
+  for (var keysIndex = 0; keysIndex < keys.length; keysIndex++) {
+    for (var property in source) {
+      if (keys[keysIndex] === property && source[property] !== undefined) {
+        pickObject[property] = source[property];
+      }
+    }
```
Comment (thebearingedge) [inline object-manipulation-2/src/pick.js:9]: This inner loop is not necessary, you can use `keys[keyIndex]` instead of your `property` variable.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by SamuelCho1** [comment]
Screen Shot 2021-02-15 at 7 38 00 PM
Screen Shot 2021-02-15 at 7 38 08 PM
Screen Shot 2021-02-15 at 7 38 13 PM

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**